pilot/pkg/bootstrap/server.go

```diff
@@ -152,6 +153,12 @@ type Server struct {
 	// fileWatcher used to watch mesh config, networks and certificates.
 	fileWatcher filewatcher.FileWatcher
 
+	// certWatcher watches the certificates for changes and triggers a notification to Istiod.
+	cacertsWatcher *fsnotify.Watcher
+	cacertsMap     map[string]bool
```

Comment:
any strong reason why we need this map at all? All the code that reads this map is on the path of execution of the goroutine that results from an fsnotify change. Can this not be maintained on the stack? Also, IMO, as discussed in the threads below, its okay to say that any valid fsnotify change on any of the files should refresh the certBundle, without going to deep into which particular file has been modified.

Reply:
Yes, make sense.